File: apps/sqs/golang/consumer/consumer.go
package main

import (
	"context"
	"encoding/json"
	"fmt"

	ddlambda "github.com/DataDog/datadog-lambda-go"
	"github.com/DataDog/sample-apps/apps/sqs/golang/internal"
	"github.com/aws/aws-lambda-go/events"
	"github.com/aws/aws-lambda-go/lambda"
)

func consumer(ctx context.Context, event events.SQSEvent) (string, error) {
	traceID := internal.TraceID(ctx)
	for _, record := range event.Records {
		var msg internal.Message
		fmt.Printf("received sqs message %s", record.Body)
		json.Unmarshal([]byte(record.Body), &msg)
		ddlambda.Metric(
			"trace_context.propagated.sqs", 1,
			fmt.Sprintf("consumer_runtime:%s", internal.Runtime),
			fmt.Sprintf("producer_runtime:%s", msg.Runtime),
			fmt.Sprintf("success:%t", traceID == msg.TraceID),
			"transport:sqs",
		)
	}
	return "ok", nil
}

func main() {
	lambda.Start(ddlambda.WrapFunction(consumer, nil))
}

File: apps/trace-propagation/sns-sqs/java/src/main/java/example/Consumer.java
package example;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.amazonaws.services.lambda.runtime.events.SQSEvent;
import com.timgroup.statsd.NonBlockingStatsDClientBuilder;
import com.timgroup.statsd.StatsDClient;
import org.json.JSONObject;

import static example.Context.currentTraceId;
import static example.Context.runtime;

public class Consumer implements RequestHandler<SQSEvent, Void>{

  private static final StatsDClient statsd = new NonBlockingStatsDClientBuilder().hostname("localhost").build();

  @Override
  public Void handleRequest(SQSEvent event, Context context)
  {
    String traceId = currentTraceId();
    for (SQSEvent.SQSMessage message : event.getRecords()) {
      JSONObject body = new JSONObject(message.getBody());
      JSONObject msg = new JSONObject(body.optString("Message"));
      System.out.println("received sqs message " + msg);

      statsd.recordDistributionValue("trace_context.propagated.sns-sqs", 1, new String[]{
        "consumer_runtime:" + runtime,
        "producer_runtime:" + msg.optString("runtime"),
        msg.optString("trace_id").equals(traceId) ? "success:true" : "success:false",
        "transport:sns-sqs",
      });
    }
    return null;
  }
}
